src/app/components/tasks/tasks.component.spec.ts

```diff
@@ -0,0 +1,141 @@
+  });
+
+  it('call toggleTask(task) => call taskService.toggleTask(task)', () => {
+    const spy = spyOn(taskService, 'toggleTask').and.stub();
+    component.toggleTask(task);
+    expect(spy).toHaveBeenCalledWith(task);
+  });
+
+  it('call updateTask(task) => call taskService.updateTask(task)', () => {
+    const spy = spyOn(taskService, 'updateTask').and.stub();
+    component.updateTask(task);
+    expect(spy).toHaveBeenCalledWith(task);
+  });
+
+  it('call removeTask(task) => call taskService.removeTask(task)', () => {
+    const spy = spyOn(taskService, 'removeTask').and.stub();
+    component.removeTask(task);
+    expect(spy).toHaveBeenCalledWith(task);
+  });
+
+  it('call toggleAllTasks() => call taskService.toggleAllTasks(isChecked)', () => {
+    const spy = spyOn(taskService, 'toggleAllTasks').and.stub();
+    const isChecked = component.isAllCompleted();
+    component.toggleAllTasks();
+    expect(spy).toHaveBeenCalledWith(!isChecked);
+  });
+
+  it('call showActiveTasks() => set filter value equal to "active"', () => {
+    component.showActiveTasks();
+    const filter = component.filter;
+    expect(filter).toBe('active');
+  });
+
+  it('call showActiveTasks() => set filter value equal to "active"', () => {
+    const spy = spyOn(console, 'log').and.stub();
+    component.filter = 'qqqqq';
+    fixture.detectChanges();
+    component.showActiveTasks();
+    expect(spy).toHaveBeenCalledWith('WTF???');
+  });
+
+  it('call showActiveTasks() => active task pass the visibility checking', () => {
+    component.showActiveTasks();
+    const activeResult = component.isShow(activeTask);
+    expect(activeResult).toBe(true);
+  });
+
+  it('call showActiveTasks() => completed task do not pass the visibility checking', () => {
+    component.showActiveTasks();
+    const completedResult = component.isShow(completedTask);
+    expect(completedResult).toBe(false);
+  });
+
+  it('call showCompletedTasks() => set filter value equal to "completed"', () => {
+    component.showCompletedTasks();
+    const filter = component.filter;
+    expect(filter).toBe('completed');
+  });
+
+  it('call showCompletedTasks() => completed task pass the visibility checking', () => {
+    component.showCompletedTasks();
+    const activeResult = component.isShow(completedTask);
+    expect(activeResult).toBe(true);
+  });
+
+  it('call showCompletedTasks() => active task do not pass the visibility checking', () => {
+    component.showCompletedTasks();
+    const completedResult = component.isShow(activeTask);
+    expect(completedResult).toBe(false);
+  });
+
+  it('call showAllTasks() => set filter value equal to "all"', () => {
+    component.showAllTasks();
+    const filter = component.filter;
+    expect(filter).toBe('all');
+  });
+
+  it('call showCompletedTasks() => any task pass the visibility checking', () => {
+    component.showAllTasks();
+    const result = component.isShow(task);
+    expect(result).toBeTruthy();
```

Comment:
Change `toBeTruthy` to `toEqual(true)`. toBeTruthy - converts types.

Reply:
done